pkg/apis/core/validation/namespacedcloudprofile.go

```diff
@@ -0,0 +1,101 @@
+
+	return allErrs
+}
+
+// ValidateNamespacedCloudProfileUpdate validates a CloudProfile object before an update.
+func ValidateNamespacedCloudProfileUpdate(newProfile, oldProfile *core.NamespacedCloudProfile) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, apivalidation.ValidateObjectMetaUpdate(&newProfile.ObjectMeta, &oldProfile.ObjectMeta, field.NewPath("metadata"))...)
+	allErrs = append(allErrs, ValidateNamespacedCloudProfileSpecUpdate(&newProfile.Spec, &oldProfile.Spec, field.NewPath("spec"))...)
+	allErrs = append(allErrs, ValidateNamespacedCloudProfile(newProfile)...)
+
+	return allErrs
+}
+
+// ValidateNamespacedCloudProfileSpecUpdate validates the spec update of a CloudProfile
+func ValidateNamespacedCloudProfileSpecUpdate(oldProfile, newProfile *core.NamespacedCloudProfileSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, apivalidation.ValidateImmutableField(oldProfile.Parent, newProfile.Parent, fldPath.Child("parent"))...)
+
+	return allErrs
+}
+
+// ValidateNamespacedCloudProfileSpec validates the specification of a CloudProfile object.
+func ValidateNamespacedCloudProfileSpec(spec *core.NamespacedCloudProfileSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, validateParent(spec.Parent, fldPath.Child("parent"))...)
+
+	if spec.Kubernetes != nil {
+		allErrs = append(allErrs, validateKubernetesSettings(*spec.Kubernetes, fldPath.Child("kubernetes"))...)
+	}
+	if spec.MachineTypes != nil {
+		allErrs = append(allErrs, validateMachineImages(spec.MachineImages, fldPath.Child("machineImages"))...)
+	}
+	if spec.MachineTypes != nil {
+		allErrs = append(allErrs, validateMachineTypes(spec.MachineTypes, fldPath.Child("machineTypes"))...)
+	}
+	if spec.VolumeTypes != nil {
+		allErrs = append(allErrs, validateVolumeTypes(spec.VolumeTypes, fldPath.Child("volumeTypes"))...)
+	}
+	if spec.Regions != nil {
+		allErrs = append(allErrs, validateRegions(spec.Regions, fldPath.Child("regions"))...)
+	}
+	if spec.SeedSelector != nil {
+		allErrs = append(allErrs, metav1validation.ValidateLabelSelector(&spec.SeedSelector.LabelSelector, metav1validation.LabelSelectorValidationOptions{AllowInvalidLabelValueInSelector: true}, fldPath.Child("seedSelector"))...)
+	}
+	if spec.CABundle != nil {
+		_, err := utils.DecodeCertificate([]byte(*(spec.CABundle)))
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("caBundle"), *(spec.CABundle), "caBundle is not a valid PEM-encoded certificate"))
+		}
+	}
+
+	return allErrs
+}
+
+func validateParent(parent core.CloudProfileReference, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if parent.Kind != "CloudProfile" {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("kind"), parent.Kind, "kind must be CloudProfile"))
```

Comment:
You could use `field.NotSupported`

Reply:
Changed as of https://github.com/gardener/gardener/pull/9440/commits/bc2bd226df5058cdde6c446ccabd89b024c97d49. Thank you.